frontend/pages/queries/edit/components/DiscardDataOption/DiscardDataOption.tsx

```diff
@@ -0,0 +1,101 @@
+import Icon from "components/Icon";
+import InfoBanner from "components/InfoBanner";
+import TooltipWrapper from "components/TooltipWrapper";
+import { IConfig } from "interfaces/config";
+import { QueryLoggingOption } from "interfaces/schedulable_query";
+import React, { useState } from "react";
+import { Link } from "react-router";
+
+const baseClass = "discard-data-option";
+
+interface IDiscardDataOption {
+  appConfig?: IConfig;
+  selectedLoggingType: QueryLoggingOption;
+  discardData: boolean;
+  setDiscardData: (value: boolean) => void;
+  breakHelpText?: boolean;
+}
+
+const DiscardDataOption = ({
+  appConfig,
+  selectedLoggingType,
+  discardData,
+  setDiscardData,
+  breakHelpText = false,
+}: IDiscardDataOption) => {
+  const [forceEditDiscardData, setForceEditDiscardData] = useState(false);
+  const disable =
+    appConfig?.server_settings.query_reports_disabled && !forceEditDiscardData;
+
+  return (
+    <div className={baseClass}>
+      {["differential", "differential_ignore_removals"].includes(
+        selectedLoggingType
+      ) && (
+        <InfoBanner color="purple-bold-border">
+          <>
+            The <b>Discard data</b> setting is ignored when differential logging
+            is enabled. This <br />
+            query&apos;s results will not be saved in Fleet.
+          </>
+        </InfoBanner>
+      )}
+      <Checkbox
+        name="discardData"
+        onChange={setDiscardData}
+        value={discardData}
+        wrapperClassName={
+          disable ? `${baseClass}__disabled-discard-data-checkbox` : ""
+        }
+      >
+        <b>Discard data</b>
+      </Checkbox>
+      <div className="help-text">
+        {disable ? (
+          <>
+            This setting is ignored because query reports in Fleet have been{" "}
+            <TooltipWrapper
+              // TODO - use JSX once new tooltipwrapper is merged
+              tipContent={
+                "A Fleet administrator can enable query reports under <br />\
+                  <b>Organization settings > Advanced options > Disable  query reports</b>."
+              }
+              position="bottom"
+            >
+              {"globally disabled."}
+            </TooltipWrapper>{" "}
+            <Link
+              to={""}
+              onClick={(e: React.MouseEvent) => {
+                e.preventDefault();
+                setForceEditDiscardData(true);
+              }}
+              className={`${baseClass}__edit-anyway`}
+            >
+              <>
+                Edit anyway
+                <Icon
+                  name="chevron"
+                  direction="right"
+                  color="core-fleet-blue"
+                  size="small"
+                />
+              </>
+            </Link>
+          </>
+        ) : (
+          <>
+            The most recent results for each host will not be available in
+            Fleet.
+            {breakHelpText ? <br /> : " "}
```

Comment:
why the explicit prop for breaking text in the UI? could this better be handled by a defined width?

Reply:
It'd have to be the exact defined width to break the text at the sentence break, which could work too. Think that'd be better? This way seems cleaner and more explicit to me.

[Here's the figma for the broken version](https://www.figma.com/file/ss8dSfoueq3REJxoGaxkNw/%237766-See-latest-query-results?type=design&node-id=925%3A26504&mode=design&t=eSvJ9mW3r4D0srQG-1) and [the non-broken version](https://www.figma.com/file/ss8dSfoueq3REJxoGaxkNw/%237766-See-latest-query-results?type=design&node-id=925%3A25188&mode=design&t=eSvJ9mW3r4D0srQG-1)

Reply:
Going to leave as-is for now since we are crunched for time, since this doesn't seem like a very important ask, and since @ghernandez345 is unavailable for re-review. Can easily change if desired.